File: backend/app/models/audit.py
from pydantic import BaseModel


class AuditLogEntryRead(BaseModel):
    id: str
    timestamp: str
    actor: str | None = None
    actor_role: str | None = None
    method: str
    path: str
    status_code: int
    payload: dict | None = None


class AuditLogPage(BaseModel):
    items: list[AuditLogEntryRead]
    total: int

File: frontend/src/types/audit.ts
export interface AuditLogEntry {
  id: string
  timestamp: string
  actor: string | null
  actor_role: string | null
  method: string
  path: string
  status_code: number
  payload: Record<string, unknown> | null
}

export interface AuditLogPage {
  items: AuditLogEntry[]
  total: number
}

export interface AuditLogQuery {
  limit: number
  offset: number
  actor?: string
  path_contains?: string
}

export interface AuditLogExportQuery {
  from?: string
  to?: string
  actor?: string
  path_contains?: string
}
